File: examples/pos-square/adapter.py
import asyncio
from apgms_sdk import OnboardingClient


async def sync_square_pos() -> None:
    client = OnboardingClient('https://api.apgms.local/v1', 'demo-key')
    payload = await client.register_webhook('https://example.com/webhooks/apgms', ['migration.started', 'migration.completed'])
    print('webhook configured', payload)


if __name__ == '__main__':
    asyncio.run(sync_square_pos())

File: examples/payroll-gusto/adapter.ts
import { OnboardingClient } from '@apgms/sdk-typescript';
import type { CreateMigrationRequest } from '@apgms/sdk-typescript';

export async function syncGustoPayroll(): Promise<void> {
  const client = new OnboardingClient({
    baseUrl: process.env.APGMS_BASE_URL ?? 'https://api.apgms.local/v1',
    apiKey: process.env.APGMS_API_KEY ?? 'demo',
  });

  const payload: CreateMigrationRequest = {
    orgId: process.env.GUSTO_ORG_ID ?? '00000000-0000-0000-0000-000000000000',
    sourceSystem: 'gusto',
    targetLedger: 'netsuite',
    dryRun: true,
  };

  const response = await client.createMigration(payload);
  console.log('dry-run started', response);
}
